compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/AnonymousObjectSuperConstructorLowering.kt

```diff
@@ -78,16 +79,19 @@ private class AnonymousObjectSuperConstructorLowering(val context: JvmBackendCon
             )
         }
 
+        fun IrExpression.transform(remapping: Map<IrVariable, IrValueParameter>): IrExpression =
+            when (this) {
+                is IrConst<*> -> this
+                is IrGetValue -> IrGetValueImpl(startOffset, endOffset, remapping[symbol.owner]?.symbol ?: symbol)
+                is IrTypeOperatorCall ->
+                    IrTypeOperatorCallImpl(startOffset, endOffset, type, operator, typeOperand, argument.transform(remapping))
+                else -> IrGetValueImpl(startOffset, endOffset, addArgument(this).symbol)
+            }
+
         fun IrDelegatingConstructorCall.transform(lift: List<IrVariable>) = apply {
             val remapping = lift.associateWith { addArgument(it.initializer!!) }
-            for (i in 0 until symbol.owner.valueParameters.size) {
-                val argument = getValueArgument(i) ?: continue
-                val mapped = when (argument) {
-                    is IrConst<*> -> null
-                    is IrGetValue -> remapping[argument.symbol.owner]
-                    else -> addArgument(argument)
-                } ?: continue
-                putValueArgument(i, IrGetValueImpl(argument.startOffset, argument.endOffset, mapped.symbol))
+            for (i in symbol.owner.valueParameters.indices) {
+                putValueArgument(i, getValueArgument(i)?.transform(remapping))
```

Comment:
What means `null` value for `getValueArgument(i)`? Shouldn't it be getValueArgument(i)!!?

Reply:
defaultArgumentInjectorPhase is after anonymousObjectSuperConstructorPhase, so it could be null if the super constructor has defaults.

Reply:
Yes, they are supported.
...so just reassigning null value that is identical to original case.